packages/keyring-controller/src/KeyringController.ts

```diff
@@ -305,12 +314,22 @@ export class KeyringController extends BaseControllerV2<
       },
     });
 
-    this.#keyring = new EthKeyringController({
-      initState: state,
-      encryptor,
-      keyringBuilders,
-      cacheEncryptionKey,
-    });
+    if (cacheEncryptionKey) {
+      this.#keyring = new EthKeyringController({
+        initState: state,
+        encryptor,
+        keyringBuilders,
+        cacheEncryptionKey,
+      });
+    } else {
+      this.#keyring = new EthKeyringController({
+        initState: state,
+        encryptor,
+        keyringBuilders,
+        cacheEncryptionKey,
+      });
+    }
```

Comment:
Something weird is going on with types. TypeScript can't infer that when `cacheEncryptionKey` is set to true, the encryptor will be an `ExportableKeyEncryptor`, but this magically works when this happen in an if branch.
My understanding is that outside the if branch, `cacheEncryptionKey` will be of type boolean, not true or false, which is the case in the if branches.

I would expect this to be enforced by the constructor types, but there's something that confuses TypeScript (and me :smile:)